tvaultuiv2/src/views/private/safe/components/AddUser/index.js

```diff
@@ -109,46 +117,66 @@ const useStyles = makeStyles(() => ({
 }));
 
 const AddUser = (props) => {
-  const { handleCancelClick, handleSaveClick } = props;
+  const { handleCancelClick, handleSaveClick, username, access } = props;
   const classes = useStyles();
   const [radioValue, setRadioValue] = useState('read');
   const [searchValue, setSearchValue] = useState('');
   const [options, setOptions] = useState([]);
   const [disabledSave, setDisabledSave] = useState(true);
+  const [searchLoader, setSearchLoader] = useState(false);
   const isMobileScreen = useMediaQuery(small);
+
   useEffect(() => {
-    if (searchValue !== '') {
-      setDisabledSave(false);
-    } else {
+    setSearchValue(username);
+    setRadioValue(access);
+  }, [username, access]);
+
+  useEffect(() => {
+    if (searchValue === '' || searchValue?.length < 3) {
       setDisabledSave(true);
+    } else {
+      setDisabledSave(false);
     }
   }, [searchValue]);
 
   const handleChange = (event) => {
     setRadioValue(event.target.value);
   };
 
-  const callSearchApi = debounce(() => {
-    apiService
-      .searchUser(data)
-      .then((res) => {
-        setOptions([]);
-        if (res?.data?.values?.length > 0) {
-          res.data.values.map((item) => {
-            return setOptions((prev) => [...prev, item.userEmail]);
+  const callSearchApi = useCallback(
+    debounce(
+      (value) => {
+        setSearchLoader(true);
+        apiService
+          .getApiCall(`/vault/v2/ldap/corpusers?CorpId=${value}`)
```

Comment:
```suggestion
          .getApiCall(`/vault/v2/ldap/corpusers?CorpId=${value}`)
```

Should be part of apiServices.js

Reply:
fixed